opendevin/action/github.py

```diff
@@ -0,0 +1,156 @@
+import os
+from dataclasses import dataclass
+from opendevin.observation import Observation, AgentErrorObservation
+from opendevin.observation.message import AgentMessageObservation
+from opendevin.observation.run import CmdOutputObservation
+from opendevin.schema import ActionType
+from typing import TYPE_CHECKING
+import requests
+import random
+import string
+
+from .base import ExecutableAction
+
+if TYPE_CHECKING:
+    from opendevin.controller import AgentController
+
+
+@dataclass
+class GithubPushAction(ExecutableAction):
+    """This pushes the current branch to github.
+
+    To use this, you need to set the OPENDEVIN_GITHUB_TOKEN environment variable.
+    The agent will return a message with a URL that you can click to make a pull
+    request.
+
+    Attributes:
+        owner: The owner of the source repo
+        repo: The name of the source repo
+        branch: The branch to push
+        action: The action identifier
+    """
+
+    owner: str
+    repo: str
+    branch: str
+    action: str = ActionType.PUSH
+
+    async def run(self, controller: 'AgentController') -> Observation:
+        github_token = os.environ.get('OPENDEVIN_GITHUB_TOKEN')
```

Comment:
We could make it a config? E.g.

```python
config.get(ConfigType.OPENDEVIN_GITHUB_TOKEN)
```

so that in the future, one can set it on web GUI.

Reply:
Done